tests/test_filter_fields.py

```diff
@@ -653,19 +675,44 @@ def check_nested(document, exclude_fields, filtered_document):
 
         document = {'a': {'b': 1}, 'b': {'a': 1}}
         exclude_fields = ['a.b', 'b.a']
-        filtered_document = {}
+        filtered_document = {'a': {}, 'b': {}}
         check_nested(document, exclude_fields, filtered_document)
 
         document = {'a': {'b': {'a': {'b': 1}}}, 'c': {'a': {'b': 1}}}
         exclude_fields = ['a.b']
-        filtered_document = {'c': {'a': {'b': 1}}}
+        filtered_document = {'a': {}, 'c': {'a': {'b': 1}}}
         check_nested(document, exclude_fields, filtered_document)
 
         document = {'name': 'anna', 'name_of_cat': 'pushkin'}
         exclude_fields = ['name']
         filtered_document = {'name_of_cat': 'pushkin'}
         check_nested(document, exclude_fields, filtered_document)
 
+        update = {'$set': {'a.b': 1, 'a.c': 3, 'b': 2, 'c': {'b': 3}}}
+        exclude_fields = ['a', 'c']
+        filtered_update = {'$set': {'b': 2}}
+        check_nested(update, exclude_fields, filtered_update, op='u')
+
+        update = {'$set': {'a.b': {'c': 3, 'd': 1}, 'a.e': 1, 'a.f': 2}}
+        exclude_fields = ['a.b.c', 'a.e']
+        filtered_update = {'$set': {'a.b': {'d': 1}, 'a.f': 2}}
+        check_nested(update, exclude_fields, filtered_update, op='u')
+
+        update = {'$set': {'a.b': {'c': 3, 'd': 1}, 'a.e': 1}}
+        exclude_fields = ['a.b.c', 'a.b.d', 'a.e']
+        filtered_update = {'$set': {'a.b': {}}}
+        check_nested(update, exclude_fields, filtered_update, op='u')
+
+        update = {'$set': {'a.b.1': 1, 'a.b.2': 2, 'b': 3}}
+        exclude_fields = ['a.b']
+        filtered_update = {'$set': {'b': 3}}
+        check_nested(update, exclude_fields, filtered_update, op='u')
+
+        update = {'$set': {'a.b': {'c': 3, 'd': 1}, 'a.e': 1}}
+        exclude_fields = ['a.b', 'a.e']
+        filtered_update = None
+        check_nested(update, exclude_fields, filtered_update, op='u')
```

Comment:
That's not exactly the same. I'm asking about a test case where exclude_fields includes a top-level field, but the update spec is updating a field more than two levels deep.